include/cilantro/icp_single_transform_combined_metric.hpp

```diff
@@ -97,10 +126,15 @@ namespace cilantro {
 
         void updateEstimate() {
             transformPoints(this->transform_, src_points_, src_points_trans_);
-
             CorrespondenceSearchCombinedMetricAdaptor<CorrespondenceSearchEngineT> corr_getter_proxy(this->correspondence_search_engine_);
             TransformT tform_iter;
-            estimateTransformCombinedMetric(dst_points_, dst_normals_, src_points_trans_, corr_getter_proxy.getPointToPointCorrespondences(), point_to_point_weight_, corr_getter_proxy.getPointToPlaneCorrespondences(), point_to_plane_weight_, tform_iter, max_optimization_iterations_, optimization_convergence_tol_, point_corr_eval_, plane_corr_eval_);
+
+            if (has_source_normals_) {
+                transformNormals(this->transform_, src_normals_, src_normals_trans_);
```

Comment:
This transformation of the source normals deviates from the paper, but it makes more sense to me to use the "most aligned" normals.

Reply:
I'll have a look at the paper soon and get back on this!

Reply:
Hi Wannes,
I hope you are doing great!
I revisited this (after some time...) and I agree that rotating the normals makes more sense. I further added this (rotation) to each linearization step in 8d660ffb12992e4ab248d9ebf1b4fea640b2b632 -- please let me know if you see anything wrong with that.
Thanks! :D

Reply:
Makes sense and the implementation seems correct to me. Did you notice any effects on performance in this case?
Thanks in any case.
BTW any work planned on cilantro for the future?

Reply:
Awesome, thanks! I didn't notice any measurable regression in some brief testing.

The next "big thing" I was planning was to add a more generic PointCloud replacement (with customizable fields at compile time), but haven't managed to resume that work for a while now...